src/Image/Darkroom/GdLib.php

```diff
@@ -62,34 +64,12 @@ protected function resize(SimpleImage $image, array $options): SimpleImage
 			return $image->resize($options['width'], $options['height']);
 		}
 
-		// crop based on focus point
-		if (Focus::isFocalPoint($options['crop']) === true) {
-			// get crop coords for focal point:
-			// if image needs to be cropped, crop before resizing
-			if ($focus = Focus::coords(
-				$options['crop'],
-				$options['sourceWidth'],
-				$options['sourceHeight'],
-				$options['width'],
-				$options['height']
-			)) {
-				$image->crop(
-					$focus['x1'],
-					$focus['y1'],
-					$focus['x2'],
-					$focus['y2']
-				);
-			}
-
-			return $image->thumbnail($options['width'], $options['height']);
-		}
+		// first crop the image in the right ratio
+		$crop  = Focus::fromDarkroomOptions($options);
+		$image = $image->crop($crop->x1, $crop->y1, $crop->x2, $crop->y2);
 
-		// normal crop with crop anchor
-		return $image->thumbnail(
-			$options['width'],
-			$options['height'] ?? $options['width'],
```

Comment:
@bastianallgeier I am wondering about this old fallback... is it needed? Cause a few lines up we don|t use one for the no-crop variant. And `Darkroom::preprocess()`, which is providing the options, calls `Dimensions::height()` which always returns an integer.

Reply:
The idea behind using the height as fallback is to create a square crop if no height is specified.

Reply:
But based on my description, do you see any possibility how that key could not exist? I think there is just no way

Reply:
Sorry, now I understand what you mean. You are right. This is no longer needed here.